### PR TITLE
Minor network logging updates

### LINKED ISSUE #5573: Use boost log attributes when logging RLPX handshake messages
There are a few pieces of information which are being manually logged in each RLPX handshake message:
* Was this connection initiated by the local node or a remote node i.e. connection direction ("egress" or "ingress")
* Remote node id (pubkey)
* Remote node address (if available i.e. if Aleth is still connected to the remote node)

Example log message:

`TRACE 04-16 09:53:10 p2p  rlpx   auth to ##a979fb57…@52.16.188.185:30303`

Manually logging this data in each message makes the log statements unnecessarily long - for example: https://github.com/ethereum/aleth/blob/aecf6c9da9c8a302915a4daea7d8e78939979bd3/libp2p/RLPxHandshake.cpp#L106-L107

We should instead use boost logging attributes to automatically include this information in each log message. Details are available here: 
* https://github.com/ethereum/aleth/pull/5568#pullrequestreview-226759809
* https://github.com/ethereum/aleth/pull/5568#issuecomment-483588551

### PR DESCRIPTION
**Pull request by halfalicious**
* Update RLPxHandshake to log connection direction (egress / ingress) via boost log attribute (fix #5573)
* Log abridged block hashes on genesis block hash mismatch during Ethereum peer validation

### CODE REVIEW

**Comment by halfalicious** [comment]
Same test is failing on linux & macos:

`521 - network/net/packetsWithChangedEndpointSuite/neighbours (Failed)`

Test is known to be unreliable (see #5544 ).

Full failure log:

```
Running 1 test case...
Test Case "neighbours": 
/home/builder/project/libdevcore/concurrent_queue.h(47): fatal error: in "_T dev::concurrent_queue<_T, _QueueT>::pop(const milliseconds&) [with _T = std::vector<unsigned char>; _QueueT = std::queue<std::vector<unsigned char>, std::deque<std::vector<unsigned char>, std::allocator<std::vector<unsigned char> > > >; std::chrono::milliseconds = std::chrono::duration<long int, std::ratio<1l, 1000l> >]": boost::exception_detail::clone_impl<dev::WaitTimeout>: /home/builder/project/libdevcore/concurrent_queue.h(47): Throw in function _T dev::concurrent_queue<_T, _QueueT>::pop(const milliseconds&) [with _T = std::vector<unsigned char>; _QueueT = std::queue<std::vector<unsigned char>, std::deque<std::vector<unsigned char>, std::allocator<std::vector<unsigned char> > > >; std::chrono::milliseconds = std::chrono::duration<long int, std::ratio<1l, 1000l> >]
Dynamic exception type: boost::exception_detail::clone_impl<dev::WaitTimeout>

/home/builder/project/test/unittests/libp2p/net.cpp(1253): last checkpoint: "neighbours" test entry

*** 1 failure is detected (5 failures are expected) in the test module "Master Test Suite"
```

File: libp2p/RLPxHandshake.cpp [hunk 1]
```diff
@@ -20,10 +20,38 @@ constexpr size_t c_ackCipherSizeBytes = 210;
 constexpr size_t c_authCipherSizeBytes = 307;
 }
 
+RLPXHandshake::RLPXHandshake(Host* _host, std::shared_ptr<RLPXSocket> const& _socket)
```
Comment (gumb0) [inline libp2p/RLPxHandshake.cpp]: It looks like you could reuse the second constructor here
```
RLPXHandshake::RLPXHandshake(Host* _host, std::shared_ptr<RLPXSocket> const& _socket) :
    RLPXHandshake(_host, _socket, {})
{}

RLPXHandshake::RLPXHandshake(
    Host* _host, std::shared_ptr<RLPXSocket> const& _socket, NodeID _remote)
  : m_host(_host),
    m_remote(_remote),
    m_originated(_remote), // conversion to bool
    m_socket(_socket),
    m_idleTimer(m_socket->ref().get_io_service())
{
...
}
```

File: CHANGELOG.md [hunk 2]
```diff
@@ -8,6 +8,7 @@
 - Changed: [#5559](https://github.com/ethereum/aleth/pull/5559) Update peer validation error messages.
 - Fixed: [#5562](https://github.com/ethereum/aleth/pull/5562) Don't send header request messages to peers that haven't sent us Status yet.
 - Changed: [#5568](https://github.com/ethereum/aleth/pull/5568) Improve rlpx handshake log messages and create new rlpx log channel.
+- Changed: [#5574](https://github.com/ethereum/aleth/pull/5574) Simplify rlpx handshake logging calls and log abridged genesis hashes in peer validation output.
```
Comment (gumb0) [inline CHANGELOG.md]: Group `Changed` items together.
Although I probably wouldn't add an item for this at all, logging change is very internal and abridged hashes are minor

Comment (halfalicious) [reply CHANGELOG.md]: Agreed, changes have trivial amount of user impact. Will remove the entry for these changes.

**Pull request merged**